.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+# Source: https://github.com/actions/starter-workflows/blob/main/ci/go.yml
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  contents: read
+
+jobs:
+  # Test golangci-lint for go-version define in go.mod
+  golangci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac #v4.0.0
+      - uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe #v4.1.0
+        with:
+          go-version-file: go.mod
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc #v3.7.0
+        with:
+          version: latest
+          only-new-issues: true
+
+  tests:
+    names: Go unit tests
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Setup go
+      uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
+      with:
+        go-version-file: ./go.mod
+
+    - name: Set up gotestfmt
+      uses: GoTestTools/gotestfmt-action@v2
+
+    - name: Run tests
+      run: |
+        set -euo pipefail
+        go test -json -v ./... 2>&1 | tee /tmp/gotest.log | gotestfmt
+     
+        # Upload the original go test log as an artifact for later review.
```

Comment:
Do we really need this? We can see the output in the action logs and download it, and I don't know if we would ever go back to past tests and download the logs. Any specific reason you added this?